### DISCUSSION TITLE
Command Tabcompletion Player/Offline

### DISCUSSION THREAD

**Discussion by Ativendus** [Minestom/Minestom#1035] [Help] [answered]
Is there an argumenttype to get tabcompletion of online players and to be able to write a name of a offline player?

**Comment by EpicPlayerA10** [comment, marked as answer]
For online players you can use `ArgumentEntity` like this:
```java
ArgumentEntity onlinePlayersArg = ArgumentType.Entity("players").onlyPlayers(true);
```
Minestom does not store offline players and you have to store them yourself. When you have offline players list you can simply do:
```java
var arg = ArgumentType.Word("offline_players");
arg.setSuggestionCallback(((sender, context, suggestion) -> {
    // Return corresponding tab-completions
    suggestion.addEntry(new SuggestionEntry("some_player"));
}));
```

**Reply by ZakShearman** [reply]
What's the limit to the number of entries?

**Reply by GreatWyrm** [reply]
Just following up on EpicPlayerA10's answer, you can retrieve an offline player using `MojangUtils.fromUsername([playername])`. If you get a null from that, no player exists with that name, and you will get a JSON object if a player with that name does exist.

**Reply by EpicPlayerA10** [reply]
> What's the limit to the number of entries?

I don't know if there is a constant suggestions limit. You can try to limit suggestions to e.g. 1000

**Reply by EpicPlayerA10** [reply]
> Just following up on EpicPlayerA10's answer, you can retrieve an offline player using `MojangUtils.fromUsername([playername])`. If you get a null from that, no player exists with that name, and you will get a JSON object if a player with that name does exist.

I meant players who have logged to the server at least once.
